homeassistant/components/mqtt/__init__.py

```diff
@@ -176,7 +182,7 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         conf = dict(conf)
         hass.data[DATA_MQTT_CONFIG] = conf
 
-    if not bool(hass.config_entries.async_entries(DOMAIN)):
+    if (entry_enabled := mqtt_config_entry_enabled(hass)) is None:
```

Comment:
I'd suggest to revert this, and simplify `mqtt_config_entry_enabled` to just return a `bool`
```suggestion
    if not bool(hass.config_entries.async_entries(DOMAIN)):
```